pkg/services/rpcsrv/client_test.go

```diff
@@ -2050,10 +2050,18 @@ func testSubClientWaitWithMissedEvent(t *testing.T, local bool) {
 	b1 := blocks[0]
 	tx := b1.Transactions[0]
 
-	rcvr := make(chan *state.AppExecResult)
+	rcvr := make(chan *state.AppExecResult, 1)
+	errCh := make(chan error, 1) // Error channel for goroutine errors
+
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
+		defer wg.Done()
```

Comment:
Won't save from failure. You have this for the first (local client) test:
```
2023-04-03T07:03:16.6904087Z     client_test.go:2282: 
2023-04-03T07:03:16.6904845Z         	Error Trace:	D:\a\neo-go\neo-go\pkg\services\rpcsrv\client_test.go:2282
2023-04-03T07:03:16.6905765Z         	            				D:\a\neo-go\neo-go\pkg\services\rpcsrv\client_test.go:2088
2023-04-03T07:03:16.6906026Z         	Error:      	Condition never satisfied
2023-04-03T07:03:16.6906430Z         	Test:       	TestSubClientWaitWithMissedEvent/ws
```
Caused by slow subscriptions here:
```
	require.Eventually(t, func() bool {
		rpcSrv.subsLock.Lock()
		defer rpcSrv.subsLock.Unlock()
		return len(rpcSrv.subscribers) == 1
	}, time.Second, 100*time.Millisecond)
```
But the awaiting goroutine from this test is up and running, so it fails when the first test is already finished, and the second test is running:
```
2023-04-03T07:03:16.6918714Z panic: Fail in goroutine after TestSubClientWaitWithMissedEvent/ws has completed
2023-04-03T07:03:16.6918730Z 
2023-04-03T07:03:16.6918846Z goroutine 6871 [running]:
2023-04-03T07:03:16.6919058Z testing.(*common).Fail(0xc0050164e0)
2023-04-03T07:03:16.6919325Z 	C:/hostedtoolcache/windows/go/1.20.2/x64/src/testing/testing.go:933 +0x1a8
2023-04-03T07:03:16.6919535Z testing.(*common).Errorf(0xc0050164e0, {0x14d946a, 0x3}, {0xc006ea0000, 0x1, 0x1})
2023-04-03T07:03:16.6919775Z 	C:/hostedtoolcache/windows/go/1.20.2/x64/src/testing/testing.go:1050 +0xaf
2023-04-03T07:03:16.6920032Z github.com/stretchr/testify/assert.Fail({0x17d0ca0, 0xc0050164e0}, {0xc00781e000, 0x132}, {0x0, 0x0, 0x0})
2023-04-03T07:03:16.6920341Z 	C:/Users/runneradmin/go/pkg/mod/github.com/stretchr/testify@v1.8.0/assert/assertions.go:264 +0x412
2023-04-03T07:03:16.6921269Z github.com/stretchr/testify/assert.NoError({0x17d0ca0, 0xc0050164e0}, {0x17cd080, 0xc0076406c0}, {0x0, 0x0, 0x0})
2023-04-03T07:03:16.6921793Z 	C:/Users/runneradmin/go/pkg/mod/github.com/stretchr/testify@v1.8.0/assert/assertions.go:1392 +0x127
2023-04-03T07:03:16.6922070Z github.com/stretchr/testify/require.NoError({0x17d6750, 0xc0050164e0}, {0x17cd080, 0xc0076406c0}, {0x0, 0x0, 0x0})
2023-04-03T07:03:16.6922350Z 	C:/Users/runneradmin/go/pkg/mod/github.com/stretchr/testify@v1.8.0/require/require.go:1261 +0xa5
2023-04-03T07:03:16.6922833Z github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.testSubClientWaitWithMissedEvent.func1()
2023-04-03T07:03:16.6923589Z 	D:/a/neo-go/neo-go/pkg/services/rpcsrv/client_test.go:2275 +0x190
2023-04-03T07:03:16.6924086Z created by github.com/nspcc-dev/neo-go/pkg/services/rpcsrv.testSubClientWaitWithMissedEvent
2023-04-03T07:03:16.6924412Z 	D:/a/neo-go/neo-go/pkg/services/rpcsrv/client_test.go:2273 +0x58c
2023-04-03T07:03:16.6924702Z FAIL	github.com/nspcc-dev/neo-go/pkg/services/rpcsrv	51.189s
```
The failing goroutine is this one:
```
	go func() {
		aer, err := act.Wait(tx.Hash(), tx.ValidUntilBlock, nil)
		require.NoError(t, err)
		rcvr <- aer
	}()
```

First of all, we don't have this problem in the current master because now our server is properly shutted down. But the fix with a separate error channel is valid and can be implemented. Also, just increase waiting interval for subscriptions awaiting up to 2 seconds.